lib/autoproj/cli/version.rb

```diff
@@ -0,0 +1,62 @@
+require 'autoproj/cli/base'
+require 'rubygems'
+
+module Autoproj
+module CLI
+    class Version < Base
+        # List the version of autoproj and optionally include
+        # the installed dependencies with information about the requirement
+        # and the actual used version
+        def run(args, options = Hash.new)
+            puts "autoproj version: #{Autoproj::VERSION}"
+            if options[:deps]
+                dependency = Gem::Deprecate.skip_during do
+                    Gem::Dependency.new "autoproj", Autoproj::VERSION
+                end
+                autoproj_spec = dependency.matching_specs
+                if !autoproj_spec.empty?
+                    known_dependencies = collect_dependencies(dependency)
+                    installed_deps = {}
+                    known_dependencies.flatten.uniq do |spec|
+                        installed_deps[spec.name] = spec.version
```

Comment:
If that's your ultimate goal, why not make `collect_dependencies` return that (which would be a good alternative to the Array->Set change I suggest below)

Reply:
Updated along  these lines